src/EventListener/UserContextListener.php

```diff
@@ -146,7 +156,7 @@ public function onKernelRequest(GetResponseEvent $event)
             $response->setClientTtl($this->options['ttl']);
             $response->setVary($this->options['user_identifier_headers']);
             $response->setPublic();
-            if (4 <= Kernel::MAJOR_VERSION && 1 <= Kernel::MINOR_VERSION) {
+            if ($this->hasSessionListener && version_compare('4.1', Kernel::VERSION, '<=')) {
```

Comment:
the old code would have been broken with symfony 5.0. `<=` seems to work with all php versions: https://3v4l.org/GCltV

Reply:
The old code failed with 5.0, but works with 5.1: https://3v4l.org/PtroA

Reply:
exactly. thanks for looking at this, reminds me that i should merge it.